### PR TITLE
docs: reconcile service list with `cortex -modules`

### PR DESCRIPTION
**Pull request by jmcarp**
**What this PR does**:

Docs-only update to reconcile the list of services in architecture.md with the output of `cortex -modules`. Sorry if I missed this elsewhere, but I hadn't seen a list of all cortex services in one place in the docs.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -74,9 +74,17 @@ The Cortex services are:
 - [Ingester](#ingester)
 - [Querier](#querier)
 - [Query frontend](#query-frontend) (optional)
+- [Query scheduler](./operations/scalable-query-frontend.md#query-scheduler) (optional)
 - [Ruler](#ruler) (optional)
 - [Alertmanager](#alertmanager) (optional)
 - [Configs API](#configs-api) (optional)
+- [Compactor](./blocks-storage/compactor.md) (required for blocks storage)
```
Comment (jmcarp) [inline docs/architecture.md:81]: Just checking: is this strictly required for running blocks storage, or is it only required if using blocks storage *and* the bucket index?

Comment (bboreham) [reply docs/architecture.md:82]: The compactor also dedupes if you have replication across ingesters, and bulks up the blocks from 2 hours to 24 hours which reduces the cost of historic queries substantially.  So you can technically run without it, but nobody should do.

File: docs/architecture.md [hunk 2]
```diff
@@ -73,10 +73,19 @@ The Cortex services are:
...
 - [Configs API](#configs-api) (optional)
+- [Overrides exporter](./guides/overrides-exporter.md) (optional)
+- [Query frontend](#query-frontend) (optional)
+- [Query scheduler](./operations/scalable-query-frontend.md#query-scheduler) (optional)
+- [Ruler](#ruler) (optional)
+- [Compactor](./blocks-storage/compactor.md) (required for blocks storage)
+- [Store gateway](./blocks-storage/store-gateway.md) (required for blocks storage)
+- [Table manager](./chunks-storage/table-manager.md) (required for chunks storage)
+- [Flusher](./chunks-storage/ingesters-with-wal.md) (optional for chunks storage)
+- [Purger](./guides/deleting-series.md) (optional for chunks storage)
```
Comment (bboreham) [inline docs/architecture.md:86]: Chunks storage is deprecated; I think we should be taking it out of docs not adding more references.

**Review by bboreham — APPROVED** [review]
I've marked it as approved to speed up getting to 2 approvals, but really I don't want to add the Flusher and Purger.
Table Manager is debatable; it's quite likely someone reads about it in an older blog so we could justify mentioning it. Or equally we can say "don't add any new stuff about chunks".

File: docs/architecture.md (hunk 2 above)
Comment (jmcarp) [reply docs/architecture.md]: Makes sense. I've dropped references to services used with chunks storage.

**Review by alvinlin123 — APPROVED** [review]

**Pull request merged**